.github/workflows/icu_ci.yml

```diff
@@ -30,6 +30,47 @@ jobs:
            # Regex note: (?! ... ) is a negative lookahead. Succeed if the pattern is not present. 
           set +o pipefail && make doc 2>&1 | tee doxygen.log && ( ! grep -P 'warning:(?! .* file .?Doxyfile)' doxygen.log )
 
+  # Java7 ICU4J build and unit test
+  java7-icu4j-build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout and setup
+        uses: actions/checkout@v2
+        with:
+          lfs: true
+      - name: Checkout lfs objects
+        run: git lfs pull
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '8'
+      - name: Cache for Java 7 tarball
```

Comment:
Adding this caching step, as you've done here, and using it in the later steps, should work. I've tested that the [first run](https://github.com/echeran/icu/pull/4/checks?check_run_id=2689066843#step:5:9) will get a cache miss and the [second run](https://github.com/echeran/icu/runs/2689212122?check_suite_focus=true#step:5:14) will get a cache hit. That insulates will insulate us from the flakiness of network connections and failed downloads.